python/semantic_kernel/agents/chat_completion/chat_completion_agent.py

```diff
@@ -109,6 +124,19 @@ def __init__(
                 args["instructions"] = prompt_template_config.template
         super().__init__(**args)
 
+    @model_validator(mode="after")
+    def configure_service(self) -> "ChatCompletionAgent":
```

Comment:
what about the inverse, where someone passes Kernel, but no service? could it then resolve using kernel, either at init or at runtime?

Reply:
That's the way we handle it today, so just allowing it to proceed as before - I don't see the benefit/need to handle things in a model validator if the kernel contains the service - we'll grab it, and the execution settings at invoke time.